libs/base/include/mrpt/utils/pose_utils.h

```diff
@@ -0,0 +1,216 @@
+			quat.y(atof(curr_tokens[4].c_str()));
+			quat.z(atof(curr_tokens[5].c_str()));
+			double roll, pitch, yaw;
+			quat.rpy(roll, pitch, yaw);
+
+			p.setFromValues(
+					atof(curr_tokens[0].c_str()), // x
+					atof(curr_tokens[1].c_str()), // y
+					atof(curr_tokens[2].c_str()), // z
+					roll, pitch, yaw);
+		} // end of fromStringQuat
+
+
+		template<>
+		inline void getPoseFromString</*QUAT_REPR=*/false, /*TUM_FORMAT=*/ false>(
+				const std::string& s,
+				mrpt::poses::CPose3D& p) {
+			p.fromStringRaw(s);
+		} // end of getPoseFromString
+
+		/**Invalid form. TUM ground truth files are always in Quaternion form. */
+		template<>
+		inline void getPoseFromString<false, true>(
+				const std::string& s,
+				mrpt::poses::CPose3D& p) {
+			THROW_EXCEPTION("Invalid combination: QUAT_REPR=false, TUM_FORMAT=true");
+		}
+
+		/**\brief Specialization for strings in Quaternion form */
+		template<>
+		inline void getPoseFromString</*QUAT_REPR=*/true, /*TUM_FORMAT=*/ false>(
+				const std::string& s,
+				mrpt::poses::CPose3D& p) {
+			mrpt::poses::CPose3DQuat p_quat;
+			p_quat.fromStringRaw(s);
+			p = mrpt::poses::CPose3D(p_quat);
+		} // end of getPoseFromString
+
+
+	} // end of namespace internal
+
+
+	/**\name Parsing of textfiles with poses */
+	/**\{*/
+	/**\brief Parse the textfile and fill in the corresponding \a poses vector.
+	 *
+	 * The file to be parsed is to contain 2D or 3D poses along with their
+	 * corresponding timestamps, one line for each.
+	 *
+	 * The expected format is the following:
+	 *
+	 * For 2D Poses: timestamp x y theta (in rad)
+	 * For 3D Poses in RPY form : x y z yaw pitch roll
+	 * For 3D Poses in Quaternion form : x y z qw qx qy qz
+	 * For 3D Poses in Quaternion form [TUM Datasets] : x y z qx qy qz qw
+	 *
+	 * The 2D format abides to the groundtruth file format used by the
+	 * <em>GridMapNavSimul</em> application
+	 *
+	 * The TUM format is compatible with the groundtruth format for the
+	 * TUM RGBD datasets as generated by the * <em>rgbd_dataset2rawlog</em> MRPT
+	 * tool.
+	 *
+	 * \param[in] fname Filename from which the timestamps and poses are read
+	 * \param[out] poses_vec std::vector which is to contain the 2D poses.
+	 * \param[out] timestamps std::vector which is to contain the timestamps
+	 * for the corresponding ground truth poses. Ignore this argument if
+	 * timestamps are not needed.
+	 * \param[in] substract_init_offset If true, the filled poses are to start from
+	 * 0, that means, that if the first found pose is non-zero, it's going to be
+	 * considered and offset and substracted from all poses in the file.
+	 *
+	 * \sa http://www.mrpt.org/Collection_of_Kinect_RGBD_datasets_with_ground_truth_CVPR_TUM_2011
+	 *
+	 * \ingroup mrpt_base_grp
+	 */
+	template<class POSE_T>
+	void readFileWithPoses(
+			const std::string& fname,
+			std::vector<POSE_T>* poses_vec,
+			std::vector<mrpt::system::TTimeStamp>* timestamps=NULL,
+			bool substract_init_offset=false) {
+		MRPT_START;
```

Comment:
👍 good utility to have around!

Please, add the addition of this new function to the changelog.h file, in the part for 2.0. See plenty of examples in that same file...